lib/gollum-lib/file.rb

```diff
@@ -145,15 +147,21 @@ def get_disk_reference(name, commit)
     # that if you specify try_on_disk=true, you may or may not get a file
     # for which on_disk? is actually true.
     def find(name, version, try_on_disk = false)
-      checked = name.downcase
       map     = @wiki.tree_map_for(version)
+      if @wiki.page_file_dir
+       path = ::File.join(@wiki.page_file_dir, name)
+      else
+       path = name
+      end
+      query = ::File.join('/', path)
+
       commit  = version.is_a?(Gollum::Git::Commit) ? version : @wiki.commit_for(version)
 
-      if (result = map.detect { |entry| entry.path.downcase == checked })
-        @path    = name
+      if (result = map.detect { |entry| ::File.join('/', entry.path) == query })
```

Comment:
This seems like a costly operation to do for all entries. Is there an alternative?

Reply:
Would a regexp comparison be cheaper?

Reply:
What would the regexp be? I will run a quick benchmark.

Reply:
`/\/+#{entry.path}/ =~ query`

Reply:
```ruby
map = ["a", "/b", "/c/d"] 
query = '/b'
puts Benchmark.measure{
  50_000.times do
    map.detect{|entry| ::File.join('/', entry) == query}
  end
}
#=>   0.100000   0.010000   0.110000 (  0.112311)

puts Benchmark.measure{
  50_000.times do
    map.detect{|entry| /\/+#{entry}/ =~ query}
  end
}
#=>  0.920000   0.020000   0.940000 (  0.941059)
```

If this benchmark makes sense (@dometto ?) then straightforward string comparison is much faster than regexp and you should keep the current solution.

Reply:
Thanks! Great to actually have some data here. I think the results also make sense, as File::Join appears to be implemented entirely in C.